stage1/rootfs/prepare-app/prepare-app.c

```diff
@@ -179,5 +179,15 @@ int main(int argc, char *argv[])
 	pexit_if(symlink("/dev/pts/ptmx", to) == -1,
 		"Failed to create /dev/ptmx symlink");
 
+	/* diagexec */
+	exit_if(snprintf(to, sizeof(to), "%s/.diagexec", root) >= sizeof(to),
+		"Path too long: \"%s\"", to);
+	fd = open(to, O_WRONLY|O_CREAT|O_CLOEXEC|O_NOCTTY, 0755);
+	if (fd != -1)
+		close(fd);
+	pexit_if(mount("/diagexec", to, "bind", MS_BIND, NULL) == -1,
```

Comment:
If the application has a file named "/.diagexec", that file will not be available anymore for the application...

Are you concerned about security or about accidental name collisions?

Maybe it needs a better name with "rocket" in its name, so it is unlikely that an application would want to have a file with that name.

I can also add a suffix based on the UUID of the app. So even if the application developers want to include a file with the same name in their ACI, they would need to break sha512 first.

Similarly, Docker puts the files ``/.dockerenv`` and ``/.dockerinit``, I am not sure what they are for.